apps/aetestframework/lib/worker.ex

```diff
@@ -0,0 +1,607 @@
+  """
+  @spec naming_pre_claim(String.t()) :: :ok | :unknown_node
+  def naming_pre_claim(node_name) do
+    send_command(
+      node_name,
+      "{:ok, tx} = Account.pre_claim(\"test.aet\", <<1::256>>, 10)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(tx)")
+  end
+
+  @spec naming_claim(String.t()) :: :ok | :unknown_node
+  def naming_claim(node_name) do
+    send_command(
+      node_name,
+      "{:ok, tx} = Account.claim(\"test.aet\", <<1::256>>, 10)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(tx)")
+  end
+
+  @spec naming_update(String.t()) :: :ok | :unknown_node
+  def naming_update(node_name) do
+    send_command(
+      node_name,
+      "{:ok, tx} = Account.name_update(\"test.aet\", \"{\\\"test\\\":2}\", 10) "
+    )
+
+    send_command(node_name, "Pool.add_transaction(tx)")
+  end
+
+  @spec naming_transfer(String.t()) :: :ok | :unknown_node
+  def naming_transfer(node_name) do
+    send_command(node_name, "{transfer_to_pub, transfer_to_priv} = Keys.keypair(:sign)")
+
+    send_command(
+      node_name,
+      "{:ok, transfer} = Account.name_transfer(\"test.aet\", transfer_to_pub, 10)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(transfer)")
+  end
+
+  @spec naming_revoke(String.t()) :: :ok | :unknown_node
+  def naming_revoke(node_name) do
+    send_command(node_name, "{transfer_to_pub, transfer_to_priv} = Keys.keypair(:sign)")
+
+    send_command(
+      node_name,
+      "{:ok, spend} = Account.spend(transfer_to_pub, 15, 10, <<\"payload\">>)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(spend)")
+    mine_sync_block(node_name)
+
+    send_command(
+      node_name,
+      "next_nonce = Account.nonce(Chain.chain_state().accounts, transfer_to_pub) + 1"
+    )
+
+    send_command(
+      node_name,
+      "{:ok, revoke} = Account.name_revoke(transfer_to_pub, transfer_to_priv, \"test.aet\", 10, next_nonce)"
+    )
+
+    send_command(node_name, "Pool.add_transaction(revoke)")
+  end
+
+  @spec chainstate_naming(String.t()) :: :ok | :unknown_node
+  def chainstate_naming(node_name) do
+    send_command(node_name, "naming_state = Chain.chain_state().naming")
+    send_command(node_name, "naming_state = Chain.chain_state().naming")
+  end
+
+  @doc """
+    Gets all peers for a given node
+  """
+  @spec get_all_peers(String.t()) :: :ok
+  def get_all_peers(node_name) do
+    send_command(node_name, "peers = Peers.all_peers")
+
+    send_command(
+      node_name,
+      "peers_encoded = Enum.reduce(peers, [], fn peer, acc ->
+                                acc ++ [put_in(peer.pubkey, Base.encode32(peer.pubkey))]
+                              end)"
+    )
+
+    send_command(node_name, "{:respond_peers, peers_encoded}")
+  end
+
+  @doc """
+    Checking if the port is busy
+  """
+  def busy_port?(port) do
+    :os.cmd('lsof -i -P -n | grep -w #{port}') != []
+  end
+
+  defp update_data(state, result, node, :peers) do
```

Comment:
I think you could use 3 functions instead of having type of update as last parameter to this function.